cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -4513,6 +4523,88 @@ if(ARROW_S3)
   endif()
 endif()
 
+# ----------------------------------------------------------------------
+# ucx - communication framework for modern, high-bandwidth and low-latency networks
+
+macro(build_ucx)
+  message(STATUS "Building UCX from source")
+
+  set(UCX_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/ucx_ep-install")
+
+  # link with static ucx libraries leads to test failures, use shared libs instead
```

Comment:
Is it necessary to add ucx to `LICENSE.txt`? It's shared linked.
cc @kou

Reply:
Could you add UCX to `LICENSE.txt`?

If we may bundle it to our binary distribution, we need to refer it. It doesn't depend on static/shared link.

See also:
* https://infra.apache.org/licensing-howto.html#bundled-vs-non-bundled
* https://infra.apache.org/licensing-howto.html#binary

We'll not bundle UCX for `.deb` package because Debian provides a package for UCX. But we may bundle UCX for `.rpm` package (and `.wheel`?).

Reply:
Done